### ISSUE TITLE
[BUG] TLS Version does not change on tls_reload

### ISSUE THREAD

**Issue by digipigeon** [OpenSIPS/opensips#2181]
**OpenSIPS version you are running**

```
version: opensips 3.0.3 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 4f17d052e
main.c compiled on  with gcc 7
```

**Describe the bug**

When loading TLS (mgm) from a database, if the METHOD is changed and reloads, it does not take effect.

**To Reproduce**

1. Set tls record in the database to TLSv1
2. Load OpenSIPS
3. Check (external request) and confirm TLS version is TLS 1.0
4. Change database to TLSv1_2
5. Run `tls_reload` from mi.
6. Check (external request) and confirm TLS version is still TLS 1.0

**Expected behavior**

Once changing the METHOD from the database and reloading, the active version should be changed and should match what is visible on `tls_list`

**Relevant System Logs**

**OS/environment information**
 - Operating System: 
 - OpenSIPS installation: 
 - other relevant information:
Ubuntu 18.04
Built from nightly packages.

**Additional context**

**Comment by rvlad-patrascu** [comment]
Hi @digipigeon ,

The issue doesn't seem to replicate for me. Have you also tried to print the `$tls_version` variable ? Do you use any other tls_mgm domains (DB or script defined) ?

**Comment by digipigeon** [comment]
Hi I have just repeated my test, I can confirm the following

I added the following code
```
route{
        xlog("L_WARN", "TLS VERSION: $tls_version");
```

1. Login to mi and run `tls_list`, result includes `"METHOD": "TLSv1_2",`
2. Establish a SIP Registration, in the logs observe `TLS VERSION: TLSv1.2` in the logs
3. Run `tls_reload` and then `tls_list`, result includes `"METHOD": "SSLv23",`
4. Try a new registration still observe `TLS VERSION: TLSv1.2` in the logs.
5. Just incase SSLv23 is upgrading to the latest possible, I tried to change to `TLSv1` in the database, `tls_reload` and `tls_list` contains `"METHOD": "TLSv1",`
6. Still observing in the logs `TLS VERSION: TLSv1.2`

I also can confirm this from the client side as when the browsers deprecated TLS v1 a database update and reload did not activate the new protocol.

**Comment by rvlad-patrascu** [comment]
Hi @digipigeon ,

What is your openssl version? Also, are you sure that after you change the method in opensips, the client you are testing with is actually trying to open a new connection instead of just sending a new registration on the same connection with the old TLS version ?

**Comment by digipigeon** [comment]
```version: opensips 3.0.3 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 4f17d052e
main.c compiled on  with gcc 7
```

I will test again and fully restart the softphone and restart the browser.

**Comment by digipigeon** [comment]
I still believe that there is some problem here, maybe a combination of `tls_mgm` and `db_http`. 

I came to reload the certs as they were expired. Performing a `tls_reload` did not update the certs, existing expiration dates were still in place.

**Comment by rvlad-patrascu** [comment]
Hi @digipigeon 

Check out the fix from #2433 and let me know if there is still an issue with reloading the TLS version (should be fixed as well along with the certs reloading).
